backends/dpdk/dpdkArch.cpp

```diff
@@ -1006,30 +1006,44 @@ const IR::Node* CopyMatchKeysToSingleStruct::preorder(IR::Key* keys) {
     // If any key field is from different structure, put all keys in metadata
     LOG3("Visiting " << keys);
     bool copyNeeded = false;
-    const IR::Expression* firstKey = nullptr;
 
     if (!keys || keys->keyElements.size() == 0) {
         prune();
         return keys;
     }
 
-    firstKey = keys->keyElements.at(0)->expression;
+    cstring firstKeyStr = "";
+    bool firstKeyHdr = false;
+
+    if (auto firstKeyField = keys->keyElements.at(0)->expression->to<IR::Member>()) {
+        firstKeyStr = firstKeyField->expr->toString();
+        if (firstKeyStr.startsWith("h"))
```

Comment:
can you explain what 'h' means?

Reply:
A previous pass converts all header fields to the form "h.<header_name>.<field_name> and similarly metadata fields are prefixed with "m.". So, here we are checking if the match key is part of a header by checking the "h" prefix.

Reply:
That must be a comment in the code.

Reply:
Added a comment for the same.

Reply:
I liked your comment on github better than the one in the code.